modules/table/src/main/java/org/apache/ignite/internal/table/distributed/TableManager.java

```diff
@@ -638,6 +654,38 @@ private CompletableFuture<Void> waitForMetadataCompleteness(long ts) {
         return executorInclinedSchemaSyncService.waitForMetadataCompleteness(HybridTimestamp.hybridTimestamp(ts));
     }
 
+    private CompletableFuture<Boolean> onZoneReplicaCreated(PartitionReplicaEventParameters parameters) {
+        if (!PartitionReplicaLifecycleManager.ENABLED) {
+            return completedFuture(false);
+        }
+
+        List<CompletableFuture<?>> futs = new ArrayList<>();
+
+        readyToProcessTableStarts.thenRun(() -> {
+            Set<TableImpl> zoneTables = zoneTables(parameters.zoneDescriptor().id());
+
+            PartitionSet singlePartitionIdSet = PartitionSet.of(parameters.partitionId());
+
+            zoneTables.forEach(tbl -> {
+                futs.add(inBusyLockAsync(busyLock,
+                        () -> completedFuture(null)
```

Comment:
supplyAsync ?

Reply:
I think it will produce the wrong result. Actually it will produce CompleatableFuture<CompletableFuture<?>> and the further steps will be executed when the body of supplyAsync(() -> {...}) will be executed, instead of the getOrCreatePartitionStorages.

By the way, looks like we already have this mistake here https://github.com/apache/ignite-3/blob/f6816866c5c955869c626a8ad30ffb7d660fc95c/modules/table/src/main/java/org/apache/ignite/internal/table/distributed/TableManager.java#L2528 - we will return completed future from the handleChangeStableAssignmentEvent when the synchronous part of stopAndDestroyPartitionAndUpdateClients done, but the future of stopAndDestroyPartitionAndUpdateClients will be unfinished yet. @alievmirza could you please verify my arguments here?

Reply:
Oh, I see, you mean supplyAsync + thenCompose(identity()) pattern. Fixed.